r/configure

```diff
@@ -252,6 +252,10 @@ echo "#include $PKG_TEST_HEADER" | ${TEST_CMD} >/dev/null 2>&1
 
 if [ $? -eq 0 ]; then
   # Check for features
+  if [ "$LIB_DIR" = "" ]; then
+    # In case LIB_DIR wasn't previously set, infer from PKG_DIRS
+    LIB_DIR=`echo $PKG_DIRS | sed -e 's/^-L//'`
+  fi
```

Comment:
I think that the following approach (we must set `LIB_DIR` when `PKG_*` are set) is better than this approach:

```diff
diff --git a/r/configure b/r/configure
index 65528bdc3..f452fffb1 100755
--- a/r/configure
+++ b/r/configure
@@ -126,10 +126,11 @@ else
     if [ "$UNAME" = "Darwin" ] && [ "$FORCE_BUNDLED_BUILD" != "true" ]; then
       if [ "$FORCE_AUTOBREW" != "true" ] && [ "`command -v brew`" ] && [ "`brew ls --versions ${PKG_BREW_NAME}`" != "" ]; then
         echo "*** Using Homebrew ${PKG_BREW_NAME}"
-        BREWDIR=`brew --prefix`
+        BREWDIR=`brew --prefix ${PKG_BREW_NAME}`
+        LIB_DIR="$BREWDIR/lib"
         PKG_LIBS="-larrow -larrow_bundled_dependencies"
-        PKG_DIRS="-L$BREWDIR/opt/$PKG_BREW_NAME/lib $PKG_DIRS"
-        PKG_CFLAGS="-I$BREWDIR/opt/$PKG_BREW_NAME/include $PKG_CFLAGS"
+        PKG_DIRS="-L$LIB_DIR $PKG_DIRS"
+        PKG_CFLAGS="-I$BREWDIR/include $PKG_CFLAGS"
       else
         echo "*** Downloading ${PKG_BREW_NAME}"
         if [ -f "autobrew" ]; then
@@ -144,7 +145,9 @@ else
         if [ $? -ne 0 ]; then
           echo "Failed to retrieve binary for ${PKG_BREW_NAME}"
         fi
-        # autobrew sets `PKG_LIBS`, `PKG_DIRS`, and `PKG_CFLAGS`
+        # autobrew sets `PKG_LIBS`, `PKG_DIRS`, `PKG_CFLAGS`, and `BREWDIR`
+        # but `LIB_DIR` isn't set.
+        LIB_DIR="$BREWDIR/lib"
       fi
     else
       if [ "${NOT_CRAN}" = "true" ]; then
```

Reply:
@kou is this possibly relevant? `PKG_LIBS` and `PKG_DIRS` seem to be set correctly but for some reason `LIB_DIR` isn't. 

```diff
diff --git a/r/configure b/r/configure
index 65528bdc38..8e130c53b4 100755
--- a/r/configure
+++ b/r/configure
@@ -110,7 +110,7 @@ else
     PKG_CFLAGS="$PKGCONFIG_CFLAGS $PKG_CFLAGS"
     PKG_LIBS="${PKGCONFIG_LIBS}"
     PKG_DIRS="${PKGCONFIG_DIRS}"
-    LIB_DIR=${FOUND_LIB_DIR}
+    LIB_DIR="${FOUND_LIB_DIR}"
 
     # Check for version mismatch
     PC_LIB_VERSION=`pkg-config --modversion arrow`
```

Reply:
Sorry. Could you explain more the context for it? Can I see a build log for the case? This case https://github.com/apache/arrow/issues/35140 ? Or other case?

If this is for #35140, I think that the isn't related because `FOUND_LIB_DIR` doesn't have any space:

```text
*** Arrow C++ libraries found via pkg-config at /opt/homebrew/Cellar/apache-arrow/11.0.0_3/lib
```

Reply:
Ah, #35140 uses Apache Arrow 11.0.0.
I was seeing the main branch.

Yes. The change https://github.com/apache/arrow/pull/35147#discussion_r1175945751 will fix #35140. And it's already in the main branch: https://github.com/apache/arrow/pull/34229

Reply:
From the conda-side, we were able to fix it by specifying `LIB_DIR=$PREFIX/lib` in our build script, see https://github.com/conda-forge/r-arrow-feedstock/pull/63

Reply:
🤦 my mistake, sorry I missed #34229, that explains some of the confusion. I'll tidy this PR up with any remaining useful changes and get it merged.